service/matching/matchingEngine.go

```diff
@@ -475,16 +484,18 @@ pollLoop:
 			return nil, err
 		}
 
-		var maxDispatch *float64
-		if request.TaskQueueMetadata != nil && request.TaskQueueMetadata.MaxTasksPerSecond != nil {
-			maxDispatch = &request.TaskQueueMetadata.MaxTasksPerSecond.Value
-		}
 		// Add frontend generated pollerID to context so taskqueueMgr can support cancellation of
 		// long-poll when frontend calls CancelOutstandingPoll API
 		pollerCtx := context.WithValue(hCtx.Context, pollerIDKey, pollerID)
 		pollerCtx = context.WithValue(pollerCtx, identityKey, request.GetIdentity())
 		taskQueueKind := request.TaskQueue.GetKind()
-		task, err := e.getTask(pollerCtx, taskQueue, maxDispatch, taskQueueKind)
+		pollMetadata := &pollMetadata{
+			workerVersionCapabilities: request.WorkerVersionCapabilities,
+		}
+		if request.TaskQueueMetadata != nil && request.TaskQueueMetadata.MaxTasksPerSecond != nil {
+			pollMetadata.ratePerSecond = &request.TaskQueueMetadata.MaxTasksPerSecond.Value
```

Comment:
Just FTR, this rate limiting mechanism can probably go away in favor of a user data based solution.